build/nlu-regression.js

```diff
@@ -0,0 +1,154 @@
+const fs = require('fs')
+const axios = require('axios').default
+const _ = require('lodash')
+const chalk = require('chalk')
+
+const repoRootDir = `${__dirname}/..`
+const nluTestingDir = `${repoRootDir}/modules/nlu-testing/`
+
+const BASE = 'http://localhost:3000'
+
+const BOT_ID = 'testy'
+const BOT_INFO = {
+  id: BOT_ID,
+  name: 'testy',
+  template: {
+    id: 'bp-nlu-regression-testing',
+    moduleId: 'nlu-testing'
+  }
+}
+
+const USER_CREDENTIALS = {
+  email: 'admin',
+  password: '123456'
+}
+
+const login = async () => {
+  try {
+    const { data } = await axios.post(`${BASE}/api/v1/auth/login/basic/default`, USER_CREDENTIALS)
+    return data.payload.token
+  } catch {
+    return
+  }
+}
+
+const signup = async () => {
+  try {
+    const { data } = await axios.post(`${BASE}/api/v1/auth/register/basic/default`, USER_CREDENTIALS)
+    return data.payload.token
+  } catch {
+    return
+  }
+}
+
+const createBot = async axiosConfig => {
+  try {
+    await axios.post(`${BASE}/api/v1/admin/bots`, BOT_INFO, axiosConfig)
+  } catch (err) {
+    const { status } = err.response
+    if (status === 409) {
+      console.log('bot already exists')
+      return
+    }
+    throw err
+  }
+}
+
+const waitForTraining = async axiosConfig => {
+  return new Promise(resolve => {
+    let i = 0
+    console.log(`training...`)
+    const intervalId = setInterval(async () => {
+      const { data: trainingStatus } = await axios.get(`${BASE}/api/v1/bots/${BOT_ID}/mod/nlu/train`, axiosConfig)
+
+      if (!trainingStatus.isTraining) {
+        clearInterval(intervalId)
+        resolve()
+      } else {
+        console.log(`training... ${2 * ++i}s`)
+      }
+    }, 2000)
+  })
+}
+
+const runAllTests = async axiosConfig => {
+  const baseNluTesting = `${BASE}/api/v1/bots/${BOT_ID}/mod/nlu-testing`
+  const { data: tests } = await axios.get(`${baseNluTesting}/tests`, axiosConfig)
+
+  let passedTests = 0
+  let i = 0
+  for (const test of tests) {
+    const retry = async () => {
+      const { data } = await axios.post(`${baseNluTesting}/tests/${test.id}/run`, '', axiosConfig)
+      return data
+    }
+
+    let testResult
+    try {
+      testResult = await retry()
+    } catch (err) {
+      console.error(err, 'retrying')
+      testResult = await retry()
+    }
+
+    passedTests += testResult.success ? 1 : 0
+    console.log(`(${i++} /${tests.length}) #${test.id}`, 'success: ', testResult.success)
+  }
+
+  return _.round((passedTests / tests.length) * 100, 1)
+}
+
+const compareScore = async score => {
+  const latestResultsFile = `${nluTestingDir}/src/bot-templates/bp-nlu-regression-testing/latest-results.csv`
+  const latestResultsContent = fs.readFileSync(latestResultsFile, { encoding: 'utf8' })
+  const previousScoreOccurence = latestResultsContent.match(/summary: ((100|\d{1,2})[.]\d{1})?/gm)
```

Comment:
that's some clever way of reading the summary. well done 👍

Reply:
:)